ui/panels-plugin/src/plugins/gauge-chart/gauge-chart-model.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { JsonObject } from '@perses-dev/core';
+import { GraphQueryDefinition } from '@perses-dev/plugin-system';
+import { UnitOptions } from '@perses-dev/components';
+import { ThresholdOptions } from '../../model/thresholds';
+import { CalculationType } from '../../model/calculations';
+
+/**
+ * The Options object type supported by the GaugeChart panel plugin.
+ */
+export interface GaugeChartOptions extends JsonObject {
+  query: GraphQueryDefinition;
+  calculation: CalculationType;
+  unit?: UnitOptions;
+  thresholds?: ThresholdOptions;
+  max?: number;
+}
+
+/**
+ * Creates the initial/empty options for a GaugeChart panel.
+ */
+export function createInitialGaugeChartOptions(): GaugeChartOptions {
+  return {
+    // TODO: How do you represent an initially empty/unset graph query?
```

Comment:
The stuff below itself doesn't look bad to me :)

Reply:
I think I remember my last company had a utility function for generating a default empty query too, and we used that in different places (like maybe we'd want to use between these different charts)

Reply:
That's a good idea. I've definitely seen that kinda pattern too before.